package/MDAnalysis/core/groups.py

```diff
@@ -816,6 +816,24 @@ def wrap(self, compound="atoms", center="com", box=None):
             if not all(s == 0.0):
                 o.atoms.translate(s)
 
+    def groupby(self, topattr):
+        """Obtain groupings of the components of this Group according the values
+        of a given topology attribute.
+
+        Parameters
+        ----------
+        topattr: str
+           Topology attribute to group components by.
+
+        Returns
+        -------
+        dict
+            Unique values of the topology attribute as keys, Groups as values.
+
+        """
+        return {i: self[self.__getattribute__(topattr) == i] for i in 
```

Comment:
Because we're fooling around with `__getattr__` methods in our classes, it might be safer to use `getattr(self, topattr)` which is more like calling `self.topattr` (rather than directly going to the class method).  I'm not sure if calling the class method directly could bypass something